semantics/src/main/scala/aqua/semantics/expr/func/ArrowSem.scala

```diff
@@ -53,18 +53,24 @@ class ArrowSem[S[_]](val expr: ArrowExpr[S]) extends AnyVal {
   ): Alg[Raw] = for {
     streamsInScope <- N.streamsDefinedWithinScope()
     retValues <- T.endArrowScope(expr.arrowTypeExpr)
+    // TODO: wrap with local on...via...
+    retsAndArgs = retValues zip funcArrow.codomain.toList
+    streamsThatReturnAsStreams = retsAndArgs.collect {
```

Comment:
This variable is not used as a whole, how about doing `unzip` here instead of `.map(_._n...` later?